### ISSUE TITLE
Title: Issue Regarding the Implementation of **`ResidualFSQ`**

### ISSUE THREAD

**Issue by qinxiangyujiayou** [lucidrains/vector-quantize-pytorch#166]
Dear Maintainers,

I hope this message finds you well. I'm currently working with the **`ResidualFSQ`** module and I have encountered a potential inconsistency that I'd like to discuss.

In the **`ResidualFSQ`** implementation, the residual is computed as follows:
```python
residual = first(self.layers).bound(x)
```

However, in the **`FSQ`** implementation, the `bound` function and the subsequent `quantize` function are defined as:
```python
def bound(self, z, eps: float = 1e-3):
    """Bound `z`, an array of shape (..., d)."""
    half_l = (self._levels - 1) * (1 + eps) / 2
    offset = torch.where(self._levels % 2 == 0, 0.5, 0.0)
    shift = (offset / half_l).atanh()
    return (z + shift).tanh() * half_l - offset

def quantize(self, z):
    """Quantizes z, returns quantized zhat, same shape as z."""
    quantized = round_ste(self.bound(z))
    half_width = self._levels // 2  # Renormalize to [-1, 1].
    return quantized / half_width
```

Comparing these two segments, I noticed that in the `quantize` function, the variable `quantized` is normalized by `half_width`, which is not considered in the `bound` function used in **`ResidualFSQ`**. Could this difference potentially lead to a mismatch or an unintended behavior in the module's functionality?

Could you please clarify whether this discrepancy is intentional and if it might affect the computation outcomes? Any insights or further explanations would be greatly appreciated.

Thank you for your time and assistance.

Best regards,

**Comment by lucidrains** [comment]
@qinxiangyujiayou hello! thanks for testing out this experimental work. very curious if you find the residual formulation useful at all

you are correct, but i removed it altogether because [another repo](https://github.com/lucidrains/meshgpt-pytorch) is using residual lfq successfully without any initial bounding

i tested out the residual fsq just now on the autoencoder in the example and it converges just fine.

**Comment by lucidrains** [comment]
@qinxiangyujiayou what are you training this on? image generation?
